frontend/javascripts/oxalis/model/sagas/dataset_saga.ts

```diff
@@ -31,3 +32,29 @@ export function* watchMaximumRenderableLayers(): Saga<void> {
 
   yield* takeEvery(["WK_READY", "UPDATE_LAYER_SETTING", "UPDATE_DATASET_SETTING"], warnMaybe);
 }
+
+let downsamplingWarningDidShow = false;
+export function* watchZ1Downsampling(): Saga<void> {
+  function* maybeShowWarning(): Saga<void> {
+    const currentRes = yield* select((state) => getCurrentResolution(state));
+    const currentZoomStep = yield* select((state) => state.flycam.zoomStep);
+    const minVoxelPerPixel = 0.5;
+
+    if (!downsamplingWarningDidShow) {
+      // checking the downsampled dimensions x and y
+      for (let i = 0; i < 2; i++) {
+        const voxelPerPixelXY = currentZoomStep / currentRes[i];
+        if (voxelPerPixelXY < minVoxelPerPixel) {
+          Toast.warning(messages["dataset.z1_downsampling_hint"]);
+          downsamplingWarningDidShow = false;
```

Comment:
see my other suggestion for `userClosedWarning`. if you follow that, the line here can be removed completely as the `onClose` handler will set the boolean to true.

```suggestion
```